crates/solidity/outputs/cargo/crate/src/backend/passes/p5_resolve_references.rs

```diff
@@ -81,21 +84,33 @@ impl Pass {
 
     fn enter_scope_for_node_id(&mut self, node_id: NodeId) {
         let scope_id = self.binder.scope_id_for_node_id(node_id).unwrap();
-        self.scope_stack.push(scope_id);
+        self.scope_stack.push((scope_id, scope_id));
+    }
+
+    fn enter_replace_scope_for_node_id(&mut self, node_id: NodeId) {
```

Comment:
nit: typo? `replace_scope_for_node_id`?

Reply:
I guess `replace_scope_for_node_id` is clear enough, but I'm not sure semantically is correct, as we are effectively entering or opening the given scope (via its node ID). The `replace` part refers to what happens to the stack. Maybe `enter_and_replace_scope_for_node_id`?

Reply:
I went with `replace_scope_for_node_id` as it's more similar to `replace_scope` in p2.